lib/rspec/que/queue_up.rb

```diff
@@ -1,89 +1,155 @@
 require 'rspec/mocks/argument_list_matcher'
+require 'time'
 
 module RSpec
   module Que
     module Matchers
       class QueueUp
         def initialize(job_class = nil)
+          @matchers = [QueuedSomething.new]
+          @matchers << QueuedClass.new(job_class) if job_class
           @job_class = job_class
+
+          @stages = nil
         end
 
         def matches?(block)
           @before_jobs = enqueued_jobs.dup
```

Comment:
I would not save `before_jobs` on the instance. Better pass it as parameter to `new_jobs` or inline `new_jobs`.